services/director-v2/src/simcore_service_director_v2/utils/dask.py

```diff
@@ -131,30 +135,45 @@ async def parse_output_data(
         await (await ports.outputs)[port_key].set_value(value_to_transfer)
 
 
+_PVType = Optional[_NPItemValue]
+assert len(get_args(_PVType)) == len(  # nosec
+    get_args(PortValue)
+), "Types returned by port.get_value() -> _PVType MUST map one-to-one to PortValue. See compute_input_data"
+
+
 async def compute_input_data(
     app: FastAPI,
     user_id: UserID,
     project_id: ProjectID,
     node_id: NodeID,
 ) -> TaskInputData:
+    """Retrieves values registered to the inputs of project_id/node_id"""
     ports = await _create_node_ports(
         db_engine=app.state.engine,
         user_id=user_id,
         project_id=project_id,
         node_id=node_id,
     )
     input_data = {}
+
+    port: Port
     for port in (await ports.inputs).values():
-        value_link = await port.get_value()
-        if isinstance(value_link, AnyUrl):
+        value: _PVType = await port.get_value()
+
+        # Mapping _PVType -> PortValue
+        #
+        # FIXME: add map from List[AnyUrl] -> List[FileUrl] ???
+        # FIXME: how to distinguish between a url as data (e.g a thumbnail in a data object) or a
+        #        real link to be downloaded?
+        if isinstance(value, AnyUrl):
```

Comment:
not sure I understand what these changes are implying for now. maybe we can discuss these tomorrow.

Reply:
these comments are obsolete